src/stdatamodels/jwst/_kwtool/compare.py

```diff
@@ -265,6 +277,8 @@ def _compare_definitions(k, d):
         diff["type"] = subdiff
     if subdiff := _compare_enum(k, d):
         diff["enum"] = subdiff
+    if subdiff := _compare_fits_hdu(k, d):
+        diff["fits_hdu"] = subdiff
```

Comment:
```suggestion
    if subdiff := _compare_fits_hdu(k, d):
        diff["fits_hdu"] = subdiff
```
The 2 definitions provided to this function are already matched by hdu. The `kw` in the code below is a tuple of `(HDU, KEYWORD_NAME)`:
https://github.com/spacetelescope/stdatamodels/blob/f36d8b6e6b0341ffbdaceae21fd16dc3dc0d3935/src/stdatamodels/jwst/_kwtool/compare.py#L313-L318
The [keyword definitions docs](https://stdatamodels.readthedocs.io/en/latest/jwst/kwtool/kwtool.html#keyword-definitions) have some more details.